lib/upnp/igd_service.go

```diff
@@ -50,6 +50,39 @@ type IGDService struct {
 	URL       string
 	URN       string
 	LocalIP   net.IP
+	PinholeID uint16
+}
+
+// TryAddPinhole adds an IPv6 pinhole in accordance to http://upnp.org/specs/gw/UPnP-gw-WANIPv6FirewallControl-v1-Service.pdf
+// We just use the same external and internal port
+func (s *IGDService) TryAddPinhole(ctx context.Context, protocol nat.Protocol, port int, description string, duration time.Duration) (int, error) {
+	var protoNumber int
+	if protocol == nat.TCP {
+		protoNumber = 6
+	} else if protocol == nat.UDP {
+		protoNumber = 17
+	}
```

Comment:
Else? `protoNumber` will be zero in this case, which will likely not work?

Reply:
nat.Protocol is a const type that is either TCP or UDP.

Reply:
Well in the spirit of good defensive programming... What if there is ever a third value added?  You should just make sure the rest of the function doesn't run with this variable uninitialized, e.g. return early.

Reply:
Fair enough, done